FlowCrypt/src/test/java/com/flowcrypt/email/security/pgp/PgpMsgTest.kt

```diff
@@ -265,4 +282,80 @@ class PgpMsgTest {
+
+  @Test
+  fun multipleComplexMessagesTest() {
+    val testFiles = listOf(
+      "decrypt - [enigmail] basic html-0.json",
+      "decrypt - [gnupg v2] thai text-0.json",
+      "decrypt - [gnupg v2] thai text in html-0.json",
+      "decrypt - [gpgmail] signed message will get parsed and rendered " +
+          "(though verification fails, enigmail does the same)-0.json",
+      "decrypt - protonmail - auto TOFU load matching pubkey first time-0.json",
+      "decrypt - protonmail - load pubkey into contact + verify detached msg-0.json",
+      "decrypt - protonmail - load pubkey into contact + verify detached msg-1.json",
+      "decrypt - protonmail - load pubkey into contact + verify detached msg-2.json",
+      "decrypt - [symantec] base64 german umlauts-0.json",
+      "decrypt - [thunderbird] unicode chinese-0.json",
+      "decrypt - verify encrypted+signed message-0.json",
+      "verify - Kraken - urldecode signature-0.json"
+    )
+
+    for (testFile in testFiles) {
+      checkComplexMessage(testFile)
+    }
+  }
+
+  // Use this one for debugging
+  @Test
+  fun singleComplexMessageTest() {
+    val testFile = "verify - Kraken - urldecode signature-0.json"
+    checkComplexMessage(testFile)
+  }
+
+  private fun checkComplexMessage(fileName: String) {
+    println("\n*** Processing '$fileName'")
+
+    val rootObject = JsonParser.parseString(loadComplexMessage(fileName)).asJsonObject
+    val inputMsg = Base64.getDecoder().decode(rootObject["in"].asJsonObject["mimeMsg"].asString)
+    val out = rootObject["out"].asJsonObject
+    val expectedBlocks = out["blocks"].asJsonArray
+    val from = InternetAddress(out["from"].asString)
+    val to = out["to"].asJsonArray.map { InternetAddress(it.asString) }.toTypedArray()
+    val session = Session.getInstance(Properties())
+    val mimeMessage = MimeMessage(session, inputMsg.inputStream())
+    val mimeContent = PgpMsg.decodeMimeMessage(mimeMessage)
+    val processed = PgpMsg.processDecodedMimeMessage(mimeContent)
+
+    assertEquals(1, processed.from?.size ?: 0)
+    assertEquals(from, processed.from!![0])
+    assertArrayEquals(to, processed.to)
+    assertEquals(expectedBlocks.size(), processed.blocks.size)
+
+    for (i in processed.blocks.indices) {
+      println("Checking block #$i of the '$fileName'")
+
+      val expectedBlock = expectedBlocks[i].asJsonObject
+      val expectedBlockType = MsgBlock.Type.ofSerializedName(expectedBlock["type"].asString)
+      val expectedContent = expectedBlock["content"].asString.normalizeEol()
+      val expectedComplete = expectedBlock["complete"].asBoolean
+      val actualBlock = processed.blocks[i]
+      val actualContent = (actualBlock.content ?: "").normalizeEol().removeUtf8Bom()
+
+      assertEquals(expectedBlockType, actualBlock.type)
+      assertEquals(expectedComplete, actualBlock.complete)
+      assertEquals(expectedContent, actualContent)
+
+      if (
+        actualBlock.type == MsgBlock.Type.SIGNED_TEXT
```

Comment:
can be simplified to one line 

`if (actualBlock.type in setOf(MsgBlock.Type.SIGNED_TEXT, MsgBlock.Type.SIGNED_HTML)) `

Reply:
ok